src/Reservations/custom/Invoke-AzReservationReturn.ps1

```diff
@@ -0,0 +1,185 @@
+        # SendAsync Pipeline Steps to be prepended to the front of the pipeline
+        ${HttpPipelinePrepend},
+    
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Category('Runtime')]
+        [System.Uri]
+        # The URI for the proxy server to use
+        ${Proxy},
+    
+        [Parameter(DontShow)]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Category('Runtime')]
+        [System.Management.Automation.PSCredential]
+        # Credentials for a proxy server to use for the remote call
+        ${ProxyCredential},
+    
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials},
+
+        [Parameter(ParameterSetName='PostViaIdentity', Mandatory, ValueFromPipeline)]
+        [Parameter(ParameterSetName='PostViaIdentityExpanded', Mandatory, ValueFromPipeline)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Category('Path')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Models.IReservationsIdentity]
+        # Identity Parameter
+        # To construct, see NOTES section for INPUTOBJECT properties and create a hash table.
+        ${InputObject},
+
+        [Parameter(ParameterSetName='Post', Mandatory)]
+        [Parameter(ParameterSetName='PostViaIdentity', Mandatory)]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Reservations.Models.Api20221101.IRefundRequest]
+        # The return request body.
+        ${Body}
+    )
+    
+    process {
+        try {
+            $returnParam = $null
+            # Use request body directly
+            if (($PSCmdlet.ParameterSetName -eq 'Post') -or ($PSCmdlet.ParameterSetName -eq 'PostViaIdentity')) { 
+                $returnParam = ${Body}
+
+            } elseif(($PSCmdlet.ParameterSetName -eq 'PostExpanded') -or ($PSCmdlet.ParameterSetName -eq 'PostViaIdentityExpanded')) {
+                $returnParam = @{
+                    ReservationToReturnReservationId = ${ReservationToReturnReservationId}
+                    ReservationToReturnQuantity = ${ReservationToReturnQuantity}
+                    Scope = ${Scope}
+                    ReturnReason = ${ReturnReason}
+                    SessionId = ${SessionId}
+                }
+            } else {
+                Write-Error 'Cannot find expected input parameters, please try again'
+                return
+            }
+            
+            if (($PSCmdlet.ParameterSetName -eq 'PostViaIdentity') -or ($PSCmdlet.ParameterSetName -eq 'PostViaIdentityExpanded')) {
+                $response = Az.Reservations.internal\Invoke-AzReservationReturn -InputObject $InputObject -Body $returnParam -AsJob
```

Comment:
to call cmdlets inside custom scripts, please use $PSBoundParameters instead of individual parameters, please see other custom scripts under this folder.